src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,347 @@
+                                </TableCell>
+
+                                {cols.map((col, colIndex) => {
+                                    const type = col.type === "number" ? "number" : "text";
+
+                                    const value = editingRows[index]
+                                        ? editingRows[index][col.name]
+                                        : row[col.name];
+                                    const formattedValue = formatValue(value, col, type);
+
+                                    return (
+                                        <TableCell
+                                            sx={{ ...TableCellStyleOverrides }}
+                                            align={col.align}
+                                            key={colIndex}
+                                        >
+                                            {editingRows[index] && col.editable ? (
+                                                <Input
+                                                    sx={{ ...InputStyleOverrides }}
+                                                    fullWidth={type === "text" ? true : false}
+                                                    value={value}
+                                                    type={type}
+                                                    onChange={(event) =>
+                                                        handleCellChange(
+                                                            event.target.value,
+                                                            editingRows[index],
+                                                            col
+                                                        )
+                                                    }
+                                                />
+                                            ) : (
+                                                formattedValue
+                                            )}
+                                        </TableCell>
+                                    );
+                                })}
+                            </TableRow>
+                        );
+                    })}
+                </TableBody>
+                {includeFooter && footerLabel && (
+                    <TableFooter>
+                        <TableRow hover>
+                            <TableCell align="center" sx={{ ...FooterCellStyleOverrides }}>
+                                {footerLabel}
+                            </TableCell>
+                            {cols.map((col, index) => {
+                                const type = col.type === "number" ? "number" : "text";
+                                const value = footerRow[col.name];
+                                const formattedValue = formatValue(value, col, type);
+
+                                const cell = footerRow[col.name] ? (
+                                    <TableCell
```

Comment:
Let's only define the `<TableCell>` once and put this conditional content inside it.

Reply:
Done